File: JavaScript/DetectCapital/DetectCapital.js
/**
 * @param {string} word
 * @return {boolean}
 */
// (115ms)
var detectCapitalUse = function(word) {
    return (/^([A-Z]+|[A-Z]?[a-z]+)$/).test(word);
};
// (136ms)
var detectCapitalUse2 = function(word) {
    return word == word.toUpperCase() || word.slice(1) == word.slice(1).toLowerCase();
};

File: Java/DetectCapital/DetectCapital.java
package com.phoenix.ex520;

/**
 * 这道题目甚至只需要一行代码即可。
 * @author Phoenix-Z
 *
 */
public class DetectCapital {

	public boolean detectCapitalUse(String word) {
		return word.equals(word.toUpperCase()) || word.substring(1).equals(word.substring(1).toLowerCase());
		// 或者使用正则表达式
		// return word.matches("[A-Z]+|[A-Z]?[a-z]+");
    }
	
}
